compaction/compaction_manager.cc

```diff
@@ -1138,16 +1138,23 @@ future<> compaction_manager::stop() {
     if (auto cm = std::exchange(_task_manager_module, nullptr)) {
         co_await cm->stop();
     }
-    if (_state != state::none) {
-        co_return co_await std::move(*_stop_future);
+    if (_stop_future) {
+        co_await std::move(*_stop_future);
     }
 }
 
-future<> compaction_manager::really_do_stop() {
+future<> compaction_manager::really_do_stop() noexcept {
     cmlog.info("Asked to stop");
     // Reset the metrics registry
     _metrics.clear();
-    co_await stop_ongoing_compactions("shutdown");
+    try {
+        co_await stop_ongoing_compactions("shutdown");
+    } catch (...) {
```

Comment:
Do we expect stop_ongoing_compactions() to throw? IMO it should not.

Reply:
Avi is right. I cannot remember of any usage of stop_ongoing_compactions() that requires exception to be propagated. Disabling compaction? no. Drain? no. Shutdown? no. For example, a failure in a task might cause problems in migration. Maybe nodetool stop compaction API? I don't remember if errors are propagated, but failed tasks are still stopped so returning success is still correct. Check stop_tasks().

Reply:
> Maybe nodetool stop compaction API? I don't remember if errors are propagated, but failed tasks are still stopped so returning success is still correct. Check stop_tasks().

Right.

Currently the tasks are indeed stopped, but if any of the failed, we return the error to the api caller.
I'm not sure if that's actually used by anyone (maybe dtests), but I wouldn't like to break the api.

We can add a an option to return the stopped tasks status which would be false by default and true only when called from the api layer.

Reply:
Since the nodetool documentation doesn't say anything about the returned status and the api is defined as `void`, it should be safe to just log any errors from tasks rather than returning them.